test/api.js

```diff
@@ -76,6 +76,15 @@ describe('api', function() {
       });
     });
 
+    it('should NOT compile empty data string', function(done) {
+      sass.render({
+        data: ''
+      }, function(error) {
+        assert.equal(error.message, 'Data context created with empty source string');
```

Comment:
> we are pretty thin wrapper here

This is no longer the case in 3.x now that we have custom functions and types. We need to stop thinking this way.

>I wouldn't really introduce own node-sass error messages if libsass already handles the situation

We should serve our users as best we can. If Libsass creates a poor user experience we should fill those gaps until they're resolved upstream. It's important to understand that Libsass has different goals to us for the time being.

> I am not sure that having two kinds of contexts (data and file) is a good design

I don't disagree here. I think we should still take the two inputs (maybe more in the future?) but having two separate properties for this doesn't serve a benefit to the user IMHO.

>We need some work on the libsass side to have this ironed out in my opinion.

I agree that Libsass needs to provide a better developer experience.

Long story short I emphatically disagree that this is sane behaviour.

Any message we show to the user should clearly describe what went wrong and how to resolve it. IMHO `Data context created with empty source string` does neither.